src/ops/fuelup_toolchain/export.rs

```diff
@@ -0,0 +1,331 @@
+use crate::{
+    commands::toolchain::ExportCommand,
+    constants::{FUEL_TOOLCHAIN_TOML_FILE, VALID_CHANNEL_STR},
+    path::get_fuel_toolchain_toml,
+    toolchain::{DistToolchainDescription, Toolchain},
+    toolchain_override::{self, OverrideCfg, ToolchainCfg, ToolchainOverride},
+    util::version::exec_version,
+};
+use anyhow::{bail, Result};
+use component::{self, Components};
+use forc_tracing::println_warning;
+use semver::Version;
+use std::{collections::HashMap, path::PathBuf};
+use std::{fs, io::BufRead, str::FromStr};
+use tracing::info;
+
+pub fn export(command: ExportCommand, mut reader: impl BufRead) -> Result<()> {
+    let ExportCommand {
+        name,
+        channel,
+        force,
+    } = command;
+    let mut toolchain_info_path = PathBuf::from("./").join(FUEL_TOOLCHAIN_TOML_FILE);
+    if let Some(path) = get_fuel_toolchain_toml() {
+        toolchain_info_path = path;
+        if force {
+            println_warning(&format!(
+                "Because the `--force` argument was supplied, this file will be overwritten: {}",
+                &toolchain_info_path.display(),
+            ));
+            if fs::remove_file(&toolchain_info_path).is_err() {
+                bail!("failed to remove file {}", &toolchain_info_path.display());
+            }
+        } else {
+            println_warning(&format!(
+                "There is an existing toolchain override file at {}. \
+                Do you wish to replace it with a new one? (y/N) ",
+                &toolchain_info_path.display(),
+            ));
+            let mut need_replace = String::new();
+            if reader.read_line(&mut need_replace).is_err() {
+                bail!("failed to read user input");
+            }
+            if need_replace.trim() == "y" {
+                if fs::remove_file(&toolchain_info_path).is_err() {
+                    bail!("failed to remove file {}", &toolchain_info_path.display());
+                };
+            } else {
+                bail!(
+                    "Failed to export toolchain \
+                    because a toolchain override file already exists at {}.",
+                    &toolchain_info_path.display(),
+                );
```

Comment:
```suggestion
                println_warning("Canceled toolchain export");
                return Ok(());
```

@Braqzen thoughts?

Reply:
Probably okay.

I am wondering about general tool design here. Specifically, does it make sense to print a "warning" (maybe it would be better to rename to message) and then to exit with an `Ok(())` if we choose to exit the program early.

In my mind it makes more sense to propagate a specific error all the way up and something would catch that and then handle it by ignoring it or something.

Returning with an `ok` before the intended completion of the workflow is strange to me but the user did choose to cancel so perhaps that can be considered as `ok`.

It's probably too much work to implement this change throughout so it's probably okay as it is although I would generalize to `println_message` unless we have different `println_method`s already